src/lr_cd/lr_plotting.py

```diff
@@ -11,24 +11,27 @@ def plot_lr(data, est_val):
     
     Parameters
     ----------
-    data : DataFrame
-        The observed data, with 'x' being the independent variable and 'y' being the dependent variable.
-        Both 'x' and 'y' should be continuous and of the same length. The 'x' values should match
-        those in `est_val`.
-    est_val : dataframe
-        The estimated values produced by the "lr_cd" linear regression model. This DataFrame
-        should contain two columns: 'x', which matches the 'x' from `data`, and 'y_est', which
-        contains the predicted 'y' values corresponding to each 'x'.
+    X: ndarray
+        The observed data 'x', the independent variable
+    
+    y: ndarray
+       The observed data 'y', the dependent variable. Both 'x' and 'y' should be continuous and of the same length.
+
+    weights: ndarray
+        Weights generated by 'lr_cd' function. It will be used to calculate the estimated values using the observed data 'x'.
 
     Returns
     -------
-    A PNG image named 'linear_regression_plot.png' is saved in the current directory.
+    A PNG image named 'linear_regression_plot.png' is saved.
```

Comment:
Maybe we can specify its returning a scatter plot of the observed datapoints overlayed with a line coming from the fitted weights? Should we also include a path to save as a function parameter?

Reply:
Good catch! Will updated!